### PR TITLE
Illumio Rapid Ransomware Containment: New App - v1.0.0 Release

### PR DESCRIPTION
**Pull request by dsommerville-illumio**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [x] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [x] If this is a new connector or you are adding new actions
    - [x] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

* Initial release
* The Illumio connector contains the following actions:
  * `get_traffic_analysis` - query traffic data
  * `get_ip_lists` - query IP List objects
  * `get_workloads` - query Workload objects
  * `create_ruleset` - create a policy Rule Set
  * `create_rule` - create a policy Rule
  * `create_enforcement_boundary` - create a policy Enforcement Boundary to deny traffic
  * `create_virtual_service` - create a Virtual Service object
  * `create_service_binding` - bind one or more Workloads to a Virtual Service
  * `update_enforcement_mode` - update the enforcement mode of one or more Workloads
  * `provision_objects` - provision changes to policy objects

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by mishalp-crest** [comment]
@dsommerville-illumio 
Can you please remove requirements-dev.txt and exclude_files.txt if not needed?

**Comment by dsommerville-illumio** [comment]
@mishalp-crest - I'd prefer to keep them for dev use.

Can you explain the changes you've made? The Apache license appendix is unnecessary and any release notes should probably be in a `1.0.0.md` file rather than `unreleased.md`

**Comment by mishalp-crest** [comment]
@dsommerville-illumio 
Ok!
1. Change in a license for consistency across all apps
2. This is the only way to add a release note, you can check other PRs

**Comment by dsommerville-illumio** [comment]
@mishalp-crest - thanks, makes sense. Please let me know if there are any other changes needed before the release can be merged.

**Comment by mishalp-crest** [comment]
@dsommerville-illumio, No more changes required  :)

File: illumio_connector.py [hunk 1]
```diff
@@ -0,0 +1,908 @@
...
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        rule_set = None
+
+        ret_val = self.connect_pce(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        try:
+            self.debug_print("Creating rule set '{}'".format(name))
```
Comment (alexa-phantom) [inline illumio_connector.py]: Rule set is two words here, but in the JSON, it is referred to as ruleset. Please keep it consistent

File: README.md [hunk 2]
```diff
@@ -1,9 +1,992 @@
-# Splunk> Phantom
+[comment]: # "Auto-generated SOAR connector documentation"
```
Comment (alexa-phantom) [inline README.md:1]: Can we update this auto generated content? It still references Phantom

File: README.md [hunk 3]
```diff
@@ -1,9 +1,992 @@
...
+            shown if the href value is invalid.
+
+    -   **Examples:**
+        -   Bind workloads
+            "/orgs/1/workloads/abd71956-0953-4a3f-b7f7-9aecaebbc358,/orgs/1/workloads/6ee0434b-46a8-48e3-b813-bdde9ccb1c41"
+            to virtual service
+            "/orgs/1/sec_policy/active/virtual_services/ce3387fd-703a-4068-a3d2-6e71d63068f4"
+
+              
+            .
```
Comment (alexa-phantom) [inline README.md:227]: Remove

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (mishalp-crest) [reply README.md:1]: We have added [comment]: # "Auto-generated SOAR connector documentation" in place of # Splunk> Phantom, hence Phantom won't be visible.

File: README.md (hunk 3 above)
Comment (mishalp-crest) [reply README.md:227]: Done.

**Review by alexa-phantom — APPROVED** [review]

**Review by ricardo-phantom — APPROVED** [review]

**Pull request merged**